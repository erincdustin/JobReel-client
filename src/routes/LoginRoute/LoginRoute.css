.LoginRoute {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 20px;
}
.login-image {
  position: relative;
}

.login-image > img {
  height: 320px;
  width: 320px;
  position: absolute;
  top: 0;
  left: 0;
  z-index: -1;
}

.new-user {
  grid-column-start: 3;
  grid-column-end: 7;
  text-align: center;
  color: #3C3230;
}