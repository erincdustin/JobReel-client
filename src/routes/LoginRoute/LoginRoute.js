import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import LoginForm from '../../components/LoginForm/LoginForm';
import './LoginRoute.css'

class LoginRoute extends Component {

  

  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || '/dashboard';
    history.push(destination);
  }

  render() {
    return (
      <section className="LoginRoute">
        <LoginForm onLoginSuccess={this.handleLoginSuccess}/>
        <Link className="new-user" to='/'>New User? Register Here</Link>
      </section>
    )
  }
}

export default LoginRoute;