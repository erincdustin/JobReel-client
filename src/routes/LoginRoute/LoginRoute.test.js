import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import LoginRoute from './LoginRoute'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('LoginRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><LoginRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('LoginRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(<LoginRoute />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})