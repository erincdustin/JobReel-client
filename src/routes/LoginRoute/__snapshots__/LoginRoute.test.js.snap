// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginRoute snapshot test renders the UI as expected 1`] = `
<section
  className="LoginRoute"
>
  <div
    className="login-image"
  >
    <img
      alt="login-background"
      src="login.svg"
    />
  </div>
  <LoginForm
    onLoginSuccess={[Function]}
  />
  <Link
    className="new-user"
    to="/"
  >
    New User? Register Here
  </Link>
</section>
`;
